packages/xo-server/src/xo-mixins/backups-ng/index.js

```diff
@@ -354,7 +354,7 @@ export default class BackupNg {
         },
       })
     } else {
-      await Disposable.use(app.getBackupsRemoteAdapter(remoteId), adapter => adapter.deleteVmBackup(metadataFilename))
+      await Disposable.use(app.getBackupsRemoteAdapter(remote), adapter => adapter.deleteVmBackup(metadataFilename))
```

Comment:
Where does the problem come from and is there a way to avoid it in the future?

Reply:
This issue is reveled by `xo-server-test`, only tests can avoid regressions.

Reply:
And where does the problem come from?

Reply:
`getBackupsRemoteAdapter`  requires an object as params because its cache is calculated from the remote URL.

Its problematic, because if we don't pass an object, the second call to this method will return the adapter of the precedent remote.

Reply:
Do you think we could prevent this from happening by making sure keys don't contain `undefined`?

Reply:
Where this check will be implemented?

Reply:
I don't know, maybe in `MultiKeyMap` or `disposables/deduped`?

I'm just throwing ideas.

Reply:
What if we want to observe an undefined key?

Reply:
What's the chance of this happening vs the risk of this error?

Reply:
It's not the `deduped` responsibility to ensure which key should be observed.  The user of this lib should ensure to pass the correct keys because it is the only one who knows what keys are correct.